src/Operation/ExecOperation.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Exo\Operation;
+
+use InvalidArgumentException;
+
+class ExecOperation extends AbstractOperation
+{
+    const EXEC = 'execute';
+
+    /**
+     * @var string
+     */
+    private $operation;
+
+    /**
+     * @var string|null
+     */
+    private $body;
+
+    /**
+     * ExecOperation constructor.
+     *
+     * @param string   $name
+     * @param string   $operation
+     * @param ?string   $body
+     */
+    public function __construct(string $name, string $operation, string $body = null) {
+        $this->name = $name;
+        $this->operation = $operation;
+        $this->body = $body;
+    }
+
+    /**
+     * Returns the name.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * Returns the operation.
+     *
+     * @return string
+     */
+    public function getOperation(): string
+    {
+        return $this->operation;
+    }
+
+    /**
+     * Returns the body.
+     *
+     * @return string
+     */
+    public function getBody(): string
+    {
+        return $this->body;
+    }
+
+    /**
+     * Returns a new operation by applying another operation.
+     *
+     * @param ExecOperation $operation
+     * @return ExecOperation|null
+     */
+    public function apply(ExecOperation $operation)
+    {
+        if ($operation->getName() !== $this->getName()) {
+            throw new InvalidArgumentException('Cannot apply operations for a different execution.');
+        }
+
+        if ($this->getOperation() !== self::EXEC) {
+            throw new InvalidArgumentException('Invalid/Incompatible Operation Type Specified.');
+        }
+
+        return new ExecOperation(
+            $this->getName(),
+            $this->getOperation(),
+            $operation->getBody()
```

Comment:
Would this simply overwrite the current operation?

Reply:
@joshmcrae This, in the other classes, allows for additions, udpates, etc. but yes in this case, I wonder if its a necessary method. The intent I had here would be that the body would just be overwritten in a merge, but perhaps we should have a script[] and push any updates to it? What do you think?

Reply:
@joshmcrae I think for now, simply overwriting this may be the simplest solution, I am not sure I would event allow for an apply method if it wasn't a required method. I considered comparing the body and throwing an exception if they did not match so that the exec names would need to be unique. What do you think of that idea?

Reply:
I agree that `apply()` shouldn't even exist for this kind of operation. That said, it isn't technically a requirement yet because it's not enforced by `AbstractOperation`, it's just that everywhere an operation class is used it's assumed that it has that method.

If you have a look at `History::reduce()` it's type-hinted to accept `TableOperation[]` but what's really passed into it is a bunch of different operation types. `History` assumes the use of `TableOperation` in a lot of places and hasn't been refactored to deal more generally with `AbstractOperation`.

That's an issue that needs addressing so if we can do it as part of this PR that'd be great.

I'm thinking each concrete operation class will have to be able to signal whether is supports `apply()` or not, so that we can have these kind of operations that don't compose well.

Reply:
@joshmcrae understood, let me know if my refactor hits the mark for these concerns. 
FYI All parameters will no longer hint to AbstractOperation rather, they will hint to OperationInterface as this seemed more intuitive.